src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,41 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();
+            return (List<T>) bulldozerProducer.get();
+        } else if (type == Excavator.class) {
+            MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
+            return (List<T>) excavatorProducer.get();
+        } else if (type == Truck.class) {
+            MachineProducer<Truck> truckProducer = new TruckProducer();
+            return (List<T>) truckProducer.get();
+        }
+        return List.of();
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method to store the result of `get()` method and then return it, instead of casting and returning in one line.